ingestion_server/justfile

```diff
@@ -62,12 +62,17 @@ wait-up: up
 
 # Make a cURL POST request to the service with the given data
 _curl-post data host="localhost:50281":
-    curl \
+    STATUS_CODE=$(curl \
       -X POST \
       -H 'Content-Type: application/json' \
       -d '{{ data }}' \
-      -w "\n" \
-      'http://{{ host }}/task'
+      -o /dev/stderr \
+      -w "%{http_code}" \
+      'http://{{ host }}/task'); \
+    if [ $STATUS_CODE -lt 200 ] || [ $STATUS_CODE -ge 300 ]; then \
+      echo "Status: $STATUS_CODE"; \
+      exit 1; \
+    fi
```

Comment:
I was surprised by this because I assumed cURL supported some kind of method for changing the exit code if the status was bad. It turns out there's a relatively new feature (2021) that does this: [`--fail-with-body`](https://curl.se/docs/manpage.html#--fail-with-body). It has the added benefit of still printing the response body with the failure without needing to add any more flags (or checks).

Reply:
I didn't use that because a lot of us are developing on macOS and might have an older version of cURL built-in. Installing a new version of cURL is technically an option but I experienced SSL issues when I last did that and I didn't want to force an additional system-level dependency.

Reply:
Ah, TIL that macOS is on an old version :+1:

Reply:
Not only cURL, macOS comes bundled with plenty of old utils. 

I also had problems with Git when I tried to use some new feature (can't recall what it was), didn't have it, installed Git with Brew which then refused to work with SSH proto. It was a whole thing to reset everything 😢.